Source/Core/Core/HW/Memmap.cpp

```diff
@@ -57,6 +60,73 @@ u8* m_pL1Cache;
 u8* m_pEXRAM;
 u8* m_pFakeVMEM;
 
+static u32 get_MEM1Size()
+{
+  u32 MEM1_Size;
+  // I need to initialize some stuff that is normally initialized in
+  // int main(), because this funtion will be run before any of that.
+  optparse::Values options_temp;
```

Comment:
Does default initializing `optparse::Values` actually produce any useful value, or is this essentially a more complicated way of passing an empty string to `SetUserDirectory`?

I'm rather skeptical to the whole approach of calling `SetUserDirectory`/`CreateDirectories`/`SConfig::Init` more than once.

Reply:
I did little research into the workings of SetUserDirectory.  If all it needs is an empty string in order to set up the User Directory correctly (being able to locate the Dolphin.ini), I'll do that instead.

Reply:
That depends on what you mean by correctly. If you give it an empty string, then it will locate Dolphin.ini in the location where it would be if you don't override the user directory with command-line arguments. This is fine most of the time but would be bad in the case where you do override the user directory with command-line arguments.